cpu/cc2538/radio/cc2538_rf_radio_ops.c

```diff
@@ -0,0 +1,471 @@
+
+    if (pkt_len > size) {
+        RFCORE_SFR_RFST = ISFLUSHRX;
+        return -ENOBUFS;
+    }
+
+    if (buf != NULL) {
+        rfcore_read_fifo(buf, pkt_len);
+        res = pkt_len;
+        if (info != NULL) {
+            uint8_t corr_val;
+            int8_t rssi_val;
+            rssi_val = rfcore_read_byte();
+
+            /* The number of dB above maximum sensitivity detected for the
+             * received packet */
+            info->rssi = -CC2538_RSSI_OFFSET + rssi_val + IEEE802154_RADIO_RSSI_OFFSET;
+
+            corr_val = rfcore_read_byte() & CC2538_CORR_VAL_MASK;
+
+            if (corr_val < CC2538_CORR_VAL_MIN) {
+                corr_val = CC2538_CORR_VAL_MIN;
+            }
+            else if (corr_val > CC2538_CORR_VAL_MAX) {
+                corr_val = CC2538_CORR_VAL_MAX;
+            }
+
+            /* Interpolate the correlation value between 0 - 255
+             * to provide an LQI value */
+            info->lqi = 255 * (corr_val - CC2538_CORR_VAL_MIN) /
+                              (CC2538_CORR_VAL_MAX - CC2538_CORR_VAL_MIN);
+        }
+    }
+    else {
+        res = 0;
+    }
+
+    RFCORE_SFR_RFST = ISFLUSHRX;
+
+    return res;
+}
+
+static int _confirm_cca(ieee802154_dev_t *dev)
+{
+    (void) dev;
+
+    RFCORE_XREG_RFIRQM0 |= RXPKTDONE;
+
+    return cc2538_cca_status;
+}
+
+static int _request_cca(ieee802154_dev_t *dev)
+{
+    (void) dev;
+    if (!RFCORE->XREG_FSMSTAT1bits.RX_ACTIVE) {
+        return -EINVAL;
+    }
+
+    /* Ignore baseband processing */
+    RFCORE_XREG_RFIRQM0 &= ~RXPKTDONE;
+
+    cc2538_cca = true;
+    RFCORE_SFR_RFST = ISCLEAR;
+    RFCORE_SFR_RFST = STOP;
+    RFCORE_XREG_CSPCTRL &= ~CC2538_CSP_MCU_CTRL_MASK;
+
+    /* Execute the last program */
+    RFCORE_SFR_RFST = ISSTART;
+
+    return 0;
+}
+
+static int _set_cca_threshold(ieee802154_dev_t *dev, int8_t threshold)
+{
+    (void) dev;
+    (void) threshold;
+
+    if (threshold < CC2538_RF_SENSITIVITY) {
+        return -EINVAL;
+    }
+
+    RFCORE_XREG_CCACTRL0 &= ~CC2538_CCA_THR_MASK;
+    RFCORE_XREG_CCACTRL0 |= (threshold - CC2538_RSSI_OFFSET) & CC2538_CCA_THR_MASK;
+
+    return 0;
+}
+
+static int _config_phy(ieee802154_dev_t *dev, ieee802154_phy_conf_t *conf)
+{
+    (void) dev;
+    int8_t pow = conf->pow;
+    if (pow  < OUTPUT_POWER_MIN || pow > OUTPUT_POWER_MAX) {
+        return -EINVAL;
+    }
+
+    uint8_t channel = conf->channel;
+    cc2538_set_freq(IEEE802154_CHAN2FREQ(channel));
+    cc2538_set_tx_power(pow);
+
+    return 0;
+}
+
+static int _confirm_set_trx_state(ieee802154_dev_t *dev)
+{
+    (void) dev;
+    if (RFCORE->XREG_FSMSTAT0bits.FSM_FFCTRL_STATE == FSM_STATE_RX_CALIBRATION) {
+        return -EAGAIN;
+    }
+    return 0;
+}
+
+static int _request_set_trx_state(ieee802154_dev_t *dev, ieee802154_trx_state_t state)
+{
+
+    (void) dev;
+    bool wait_sfd = RFCORE->XREG_FSMSTAT0bits.FSM_FFCTRL_STATE > 0x2 && RFCORE->XREG_FSMSTAT0bits.FSM_FFCTRL_STATE < 0x7;
```

Comment:
Magic numbers? Can a define be used here?

Reply:
oh yes! I totally forgot this one

Reply:
done!